src/go/cmd/token-vendor/repository/k8s/k8s.go

```diff
@@ -94,40 +131,43 @@ func (k *K8sRepository) LookupKey(ctx context.Context, deviceID string) (*reposi
 // already the public key section of the configmap is updated.
 func (k *K8sRepository) PublishKey(ctx context.Context, deviceID, publicKey string) error {
 	slog.Debug("publishing key", slog.String("DeviceID", deviceID))
-	cm, err := createPubKeyDeviceConfig(deviceID, publicKey)
+	cm, err := createPubKeyDeviceConfig(deviceID, k.ns, publicKey)
 	if err != nil {
 		return errors.Wrapf(err, "failed to init device configmap %q/%q", k.ns, deviceID)
 	}
 	if _, err = k.kcl.CoreV1().ConfigMaps(k.ns).Create(ctx, cm, metav1.CreateOptions{}); err == nil { // no error
-
-		slog.Info("created new configmap", slog.String("Namespace", k.ns), slog.String("ConfigMap", deviceID))
+		k.cmInformer.GetStore().Add(cm)
```

Comment:
wouldn't the informcer figure that it has been added on its own?

Reply:
Thanks, that's indeed inconsistent. I dug in and it only fails if we create a createPubKeyDeviceConfig returns a config map without a `metadata` property. But I added the warning and comment just in case to help readers.

Another possibility would be to use the cache as read-through and if we get "not found" then we read from k8s - it should work in practice but it'd break the assumptions of the PublishKey() test, and making that async seemed worse than just keeping the cache up-to-date here.